src/tersets.zig

```diff
@@ -150,3 +150,17 @@ pub fn getMaxMethodIndex() usize {
 
     return max_index;
 }
+
+/// `Point` is a point represented by `time` and `value` of datatype `time_type` and `value_type`.
+fn Point(comptime time_type: type, comptime value_type: type) type {
+    return struct { time: time_type, value: value_type };
+}
+
+/// `Segment` models a straight line segment from `start_point` to `end_point`. The `time` and
+/// `value` datatypes of the points are determined by `time_type` and `value_type`.
+fn Segment(comptime time_type: type, comptime value_type: type) type {
```

Comment:
Unclear why the value type is generic instead of always `f64` like everywhere else in the library.